src/components/CodeBlock.vue

```diff
@@ -1,36 +1,98 @@
 <template>
-  <Prism
+  <PrismEditor
+    v-model="code"
     class="code-example"
-    :class="{ 'language-none': language === 'none' ? 'language-none' : '' }"
-    :language="language !== 'none' ? language : 'javascript'"
+    :class="classNames"
+    :highlight="highlighter"
+    :line-numbers="lineNumbers"
+    :readonly="!editable"
   >
+    <!--
+      @slot Codeblock content
+    -->
     <slot />
-  </Prism>
+  </PrismEditor>
 </template>
 
 <script>
-// Base Prism import
-import 'prismjs';
-// Extra language configuration
+// Vue component for Prism
+import { PrismEditor } from 'vue-prism-editor';
+import 'vue-prism-editor/dist/prismeditor.min.css';
+// Prism core
+import { highlight, languages } from 'prismjs/components/prism-core';
+// Import supported languages
 import 'prismjs/components/prism-markup-templating';
-import 'prismjs/components/prism-jsx';
+import 'prismjs/components/prism-clike';
```

Comment:
It feels like we'll probably not need this language. I assume that not including it here will reduce our bundle size slightly, maybe remove it?

Reply:
the clike component isn't a language, it's the basis used by other language packs like PHP.  I've moved it to the top to help denote that as well as left a comment.

Reply:
Okay, it needs to be there though? We haven't needed it in the past

Reply:
This package seems to need it otherwise it breaks